internal/service/alarms/internal/resourceserver/resourceserver.go

```diff
@@ -0,0 +1,65 @@
+package resourceserver
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"os"
+	"sync"
+
+	"github.com/openshift-kni/oran-o2ims/internal/service/common/clients"
+	"github.com/openshift-kni/oran-o2ims/internal/service/common/clients/resources"
+)
+
+var (
+	// url is the URL of the resource server
+	url  string
+	once sync.Once
+)
+
+// GetResourceTypes gets the list of resource types from the resource server
+func GetResourceTypes(ctx context.Context) (*[]resources.ResourceType, error) {
+	c, err := getClient()
+	if err != nil {
+		return nil, err
+	}
+
+	ctxWithTimeout, cancel := context.WithTimeout(ctx, clients.SingleRequestTimeout)
+	defer cancel()
+
+	resp, err := c.GetResourceTypesWithResponse(ctxWithTimeout, nil)
+	if err != nil {
+		return nil, fmt.Errorf("failed to execute Get operation: %w", err)
+	}
+
+	if resp.StatusCode() != http.StatusOK {
+		return nil, fmt.Errorf("status code different from 200: %d", resp.StatusCode())
+	}
+
+	return resp.JSON200, nil
+}
+
+// getClient gets the client for the resource server
+func getClient() (*resources.ClientWithResponses, error) {
+	var err error
+	once.Do(func() {
+		// TODO: update the URL to the correct one once the service is deployed
+		url = os.Getenv("RESOURCE_SERVER_URL")
+		if url == "" {
+			err = fmt.Errorf("environment variable RESOURCE_SERVER_URL must be set")
+		}
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	// Default transport (global var within net/http package) is used here. TCP connections are reused.
+	// Minimal overhead is introduced by creating a new client for each request.
+	hc := http.Client{}
```

Comment:
The resource server has HTTPS enabled so the transport will have to load the Service CA from a file.  It would be best if that was not done on each request, but rather once when the process loads.

Reply:
will review this based on previous comment

Reply:
updated